api/routes/locations-metadata.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+import Boom from 'boom';
+import config from 'config';
+import _ from 'lodash';
+
+import { db } from '../services/db';
+import metadataSchema from '../../lib/location-metadata-schema';
+
+const { strategy: authStrategy } = config.get('auth');
+
+async function checkLocation (id) {
+  // Check if the location exists.
+  const location = await db('locations')
+    .select('id')
+    .where('id', id)
+    .first();
+
+  return !!location;
+}
+
+/**
+ * @api {get} /locations/metadata GET
+ * @apiGroup Locations
+ */
+module.exports = [
+  {
+    method: ['GET'],
+    path: '/v1/locations/{id}/metadata',
+    config: {
+      description: 'Metadata associated with a given location'
+    },
+    handler: async function (request, reply) {
```

Comment:
Ideally this endpoint would also include the `locations` record for this `id`. This is useful when visiting the metadata view/edit pages directly and need the contextual location information.

Reply:
implemented